### PR TITLE
Support the EVM as a backend target.

### PR DESCRIPTION
**Pull request by otrho**
Authored by @jtriley-eth.  [Here](https://github.com/FuelLabs/sway-rfcs/blob/evm/rfcs/0007-evm-support.md)

Here's a proposal for supporting the EVM.  It still asks a few questions rather than providing a concrete path forward but that's what RFC PRs are for.

### CODE REVIEW

**Review by JoshuaBatty — APPROVED** [review]

File: rfcs/0007-evm-support.md [hunk 1]
```diff
@@ -0,0 +1,741 @@
...
+[project]
+authors = ["user"]
+entry = "main.sw"
+license = "Apache-2.0"
+name = "example"
+target = "eofv1"
+```
+
+Development, Testing, and Deployment can be done through Hardhat or Foundry using a simple plugin
+for each.
```
Comment (kayagokalp) [inline rfcs/0007-evm-support.md:117]: Hmm this makes me think if we are planning to support sway unit tests for EVM backend cc @mitchmindtree 

Currently `forc-test` is coupled with `FuelVM`, if we are going to be supporting tests for EVM as well we might need to think about it a little.

**Review by kayagokalp — APPROVED** [review]

File: rfcs/0007-evm-support.md (hunk 1 above)
Comment (mitchmindtree) [reply rfcs/0007-evm-support.md:117]: Yeah I think to start we'd want to find a nice embeddable EVM interpreter - I'm guessing there might be a lib from foundry we could plug-in for this?

Comment (otrho) [reply rfcs/0007-evm-support.md:117]: We already have an EVM testing framework set up for the E2Es which might work for unit tests too.  Not sure, @tritao looked at it initially.

Comment (kayagokalp) [reply rfcs/0007-evm-support.md:117]: Nice 👌, it will be a good reference. Thanks!

**Review by SwayStar123 — APPROVED** [review]
Ship it

**Comment by Braqzen** [comment]
> Ship it

Box it and transport it

**Pull request merged**